File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-shared-5-4/ExerciseProcedure.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class ExerciseProcedure : ISerialized
    {
        public ExerciseProcedure(XmlNode xmlNode)
        {
            XmlNode manualExerciseNode = xmlNode.SelectSingleNode("manualExercise");
            
            if (manualExerciseNode != null)
            {
                if (manualExerciseNode.Attributes["href"] != null || manualExerciseNode.Attributes["id"] != null) 
                {
                    if (manualExerciseNode.Attributes["id"] != null) 
                    {
                        manualExerciseIDRef_ = manualExerciseNode.Attributes["id"].Value;
                        ManualExercise ob = new ManualExercise(manualExerciseNode);
                        IDManager.SetID(manualExerciseIDRef_, ob);
                    }
                    else if (manualExerciseNode.Attributes["href"] != null)
                    {
                        manualExerciseIDRef_ = manualExerciseNode.Attributes["href"].Value;
                    }
                    else
                    {
                        manualExercise_ = new ManualExercise(manualExerciseNode);
                    }
                }
                else
                {
                    manualExercise_ = new ManualExercise(manualExerciseNode);
                }
            }
            
        
            XmlNode automaticExerciseNode = xmlNode.SelectSingleNode("automaticExercise");
            
            if (automaticExerciseNode != null)
            {
                if (automaticExerciseNode.Attributes["href"] != null || automaticExerciseNode.Attributes["id"] != null) 
                {
                    if (automaticExerciseNode.Attributes["id"] != null) 
                    {
                        automaticExerciseIDRef_ = automaticExerciseNode.Attributes["id"].Value;
                        AutomaticExercise ob = new AutomaticExercise(automaticExerciseNode);
                        IDManager.SetID(automaticExerciseIDRef_, ob);
                    }
                    else if (automaticExerciseNode.Attributes["href"] != null)
                    {
                        automaticExerciseIDRef_ = automaticExerciseNode.Attributes["href"].Value;
                    }
                    else
                    {
                        automaticExercise_ = new AutomaticExercise(automaticExerciseNode);
                    }
                }
                else
                {
                    automaticExercise_ = new AutomaticExercise(automaticExerciseNode);
                }
            }
            
        
            XmlNode followUpConfirmationNode = xmlNode.SelectSingleNode("followUpConfirmation");
            
            if (followUpConfirmationNode != null)
            {
                if (followUpConfirmationNode.Attributes["href"] != null || followUpConfirmationNode.Attributes["id"] != null) 
                {
                    if (followUpConfirmationNode.Attributes["id"] != null) 
                    {
                        followUpConfirmationIDRef_ = followUpConfirmationNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(followUpConfirmationNode);
                        IDManager.SetID(followUpConfirmationIDRef_, ob);
                    }
                    else if (followUpConfirmationNode.Attributes["href"] != null)
                    {
                        followUpConfirmationIDRef_ = followUpConfirmationNode.Attributes["href"].Value;
                    }
                    else
                    {
                        followUpConfirmation_ = new XsdTypeBoolean(followUpConfirmationNode);
                    }
                }
                else
                {
                    followUpConfirmation_ = new XsdTypeBoolean(followUpConfirmationNode);
                }
            }
            
        
            XmlNode limitedRightToConfirmNode = xmlNode.SelectSingleNode("limitedRightToConfirm");
            
            if (limitedRightToConfirmNode != null)
            {
                if (limitedRightToConfirmNode.Attributes["href"] != null || limitedRightToConfirmNode.Attributes["id"] != null) 
                {
                    if (limitedRightToConfirmNode.Attributes["id"] != null) 
                    {
                        limitedRightToConfirmIDRef_ = limitedRightToConfirmNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(limitedRightToConfirmNode);
                        IDManager.SetID(limitedRightToConfirmIDRef_, ob);
                    }
                    else if (limitedRightToConfirmNode.Attributes["href"] != null)
                    {
                        limitedRightToConfirmIDRef_ = limitedRightToConfirmNode.Attributes["href"].Value;
                    }
                    else
                    {
                        limitedRightToConfirm_ = new XsdTypeBoolean(limitedRightToConfirmNode);
                    }
                }
                else
                {
                    limitedRightToConfirm_ = new XsdTypeBoolean(limitedRightToConfirmNode);
                }
            }
            
        
            XmlNode splitTicketNode = xmlNode.SelectSingleNode("splitTicket");
            
            if (splitTicketNode != null)
            {
                if (splitTicketNode.Attributes["href"] != null || splitTicketNode.Attributes["id"] != null) 
                {
                    if (splitTicketNode.Attributes["id"] != null) 
                    {
                        splitTicketIDRef_ = splitTicketNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(splitTicketNode);
                        IDManager.SetID(splitTicketIDRef_, ob);
                    }
                    else if (splitTicketNode.Attributes["href"] != null)
                    {
                        splitTicketIDRef_ = splitTicketNode.Attributes["href"].Value;
                    }
                    else
                    {
                        splitTicket_ = new XsdTypeBoolean(splitTicketNode);
                    }
                }
                else
                {
                    splitTicket_ = new XsdTypeBoolean(splitTicketNode);
                }
            }
            
        
        }
        
    
        #region ManualExercise_
        private ManualExercise manualExercise_;
        public ManualExercise ManualExercise_
        {
            get
            {
                if (this.manualExercise_ != null)
                {
                    return this.manualExercise_; 
                }
                else if (this.manualExerciseIDRef_ != null)
                {
                    manualExercise_ = IDManager.getID(manualExerciseIDRef_) as ManualExercise;
                    return this.manualExercise_; 
                }
                else
                {
                      return this.manualExercise_; 
                }
            }
            set
            {
                if (this.manualExercise_ != value)
                {
                    this.manualExercise_ = value;
                }
            }
        }
        #endregion
        
        public string manualExerciseIDRef_ { get; set; }
        #region AutomaticExercise_
        private AutomaticExercise automaticExercise_;
        public AutomaticExercise AutomaticExercise_
        {
            get
            {
                if (this.automaticExercise_ != null)
                {
                    return this.automaticExercise_; 
                }
                else if (this.automaticExerciseIDRef_ != null)
                {
                    automaticExercise_ = IDManager.getID(automaticExerciseIDRef_) as AutomaticExercise;
                    return this.automaticExercise_; 
                }
                else
                {
                      return this.automaticExercise_; 
                }
            }
            set
            {
                if (this.automaticExercise_ != value)
                {
                    this.automaticExercise_ = value;
                }
            }
        }
        #endregion
        
        public string automaticExerciseIDRef_ { get; set; }
        #region FollowUpConfirmation_
        private XsdTypeBoolean followUpConfirmation_;
        public XsdTypeBoolean FollowUpConfirmation_
        {
            get
            {
                if (this.followUpConfirmation_ != null)
                {
                    return this.followUpConfirmation_; 
                }
                else if (this.followUpConfirmationIDRef_ != null)
                {
                    followUpConfirmation_ = IDManager.getID(followUpConfirmationIDRef_) as XsdTypeBoolean;
                    return this.followUpConfirmation_; 
                }
                else
                {
                      return this.followUpConfirmation_; 
                }
            }
            set
            {
                if (this.followUpConfirmation_ != value)
                {
                    this.followUpConfirmation_ = value;
                }
            }
        }
        #endregion
        
        public string followUpConfirmationIDRef_ { get; set; }
        #region LimitedRightToConfirm_
        private XsdTypeBoolean limitedRightToConfirm_;
        public XsdTypeBoolean LimitedRightToConfirm_
        {
            get
            {
                if (this.limitedRightToConfirm_ != null)
                {
                    return this.limitedRightToConfirm_; 
                }
                else if (this.limitedRightToConfirmIDRef_ != null)
                {
                    limitedRightToConfirm_ = IDManager.getID(limitedRightToConfirmIDRef_) as XsdTypeBoolean;
                    return this.limitedRightToConfirm_; 
                }
                else
                {
                      return this.limitedRightToConfirm_; 
                }
            }
            set
            {
                if (this.limitedRightToConfirm_ != value)
                {
                    this.limitedRightToConfirm_ = value;
                }
            }
        }
        #endregion
        
        public string limitedRightToConfirmIDRef_ { get; set; }
        #region SplitTicket_
        private XsdTypeBoolean splitTicket_;
        public XsdTypeBoolean SplitTicket_
        {
            get
            {
                if (this.splitTicket_ != null)
                {
                    return this.splitTicket_; 
                }
                else if (this.splitTicketIDRef_ != null)
                {
                    splitTicket_ = IDManager.getID(splitTicketIDRef_) as XsdTypeBoolean;
                    return this.splitTicket_; 
                }
                else
                {
                      return this.splitTicket_; 
                }
            }
            set
            {
                if (this.splitTicket_ != value)
                {
                    this.splitTicket_ = value;
                }
            }
        }
        #endregion
        
        public string splitTicketIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-shared-5-4/ExerciseProcedure.cpp
// ExerciseProcedure.cpp 
#include "ExerciseProcedure.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

ExerciseProcedure::ExerciseProcedure(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //manualExerciseNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* manualExerciseNode = xmlNode->FirstChildElement("manualExercise");

   if(manualExerciseNode){manualExerciseIsNull_ = false;}
   else{manualExerciseIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- manualExerciseNode , address : " << manualExerciseNode << std::endl;
   #endif
   if(manualExerciseNode)
   {
      if(manualExerciseNode->Attribute("href") || manualExerciseNode->Attribute("id"))
      {
          if(manualExerciseNode->Attribute("id"))
          {
             manualExerciseIDRef_ = manualExerciseNode->Attribute("id");
             manualExercise_ = boost::shared_ptr<ManualExercise>(new ManualExercise(manualExerciseNode));
             manualExercise_->setID(manualExerciseIDRef_);
             IDManager::instance().setID(manualExerciseIDRef_,manualExercise_);
          }
          else if(manualExerciseNode->Attribute("href")) { manualExerciseIDRef_ = manualExerciseNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { manualExercise_ = boost::shared_ptr<ManualExercise>(new ManualExercise(manualExerciseNode));}
   }

   //automaticExerciseNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* automaticExerciseNode = xmlNode->FirstChildElement("automaticExercise");

   if(automaticExerciseNode){automaticExerciseIsNull_ = false;}
   else{automaticExerciseIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- automaticExerciseNode , address : " << automaticExerciseNode << std::endl;
   #endif
   if(automaticExerciseNode)
   {
      if(automaticExerciseNode->Attribute("href") || automaticExerciseNode->Attribute("id"))
      {
          if(automaticExerciseNode->Attribute("id"))
          {
             automaticExerciseIDRef_ = automaticExerciseNode->Attribute("id");
             automaticExercise_ = boost::shared_ptr<AutomaticExercise>(new AutomaticExercise(automaticExerciseNode));
             automaticExercise_->setID(automaticExerciseIDRef_);
             IDManager::instance().setID(automaticExerciseIDRef_,automaticExercise_);
          }
          else if(automaticExerciseNode->Attribute("href")) { automaticExerciseIDRef_ = automaticExerciseNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { automaticExercise_ = boost::shared_ptr<AutomaticExercise>(new AutomaticExercise(automaticExerciseNode));}
   }

   //followUpConfirmationNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* followUpConfirmationNode = xmlNode->FirstChildElement("followUpConfirmation");

   if(followUpConfirmationNode){followUpConfirmationIsNull_ = false;}
   else{followUpConfirmationIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- followUpConfirmationNode , address : " << followUpConfirmationNode << std::endl;
   #endif
   if(followUpConfirmationNode)
   {
      if(followUpConfirmationNode->Attribute("href") || followUpConfirmationNode->Attribute("id"))
      {
          if(followUpConfirmationNode->Attribute("id"))
          {
             followUpConfirmationIDRef_ = followUpConfirmationNode->Attribute("id");
             followUpConfirmation_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(followUpConfirmationNode));
             followUpConfirmation_->setID(followUpConfirmationIDRef_);
             IDManager::instance().setID(followUpConfirmationIDRef_,followUpConfirmation_);
          }
          else if(followUpConfirmationNode->Attribute("href")) { followUpConfirmationIDRef_ = followUpConfirmationNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { followUpConfirmation_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(followUpConfirmationNode));}
   }

   //limitedRightToConfirmNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* limitedRightToConfirmNode = xmlNode->FirstChildElement("limitedRightToConfirm");

   if(limitedRightToConfirmNode){limitedRightToConfirmIsNull_ = false;}
   else{limitedRightToConfirmIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- limitedRightToConfirmNode , address : " << limitedRightToConfirmNode << std::endl;
   #endif
   if(limitedRightToConfirmNode)
   {
      if(limitedRightToConfirmNode->Attribute("href") || limitedRightToConfirmNode->Attribute("id"))
      {
          if(limitedRightToConfirmNode->Attribute("id"))
          {
             limitedRightToConfirmIDRef_ = limitedRightToConfirmNode->Attribute("id");
             limitedRightToConfirm_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(limitedRightToConfirmNode));
             limitedRightToConfirm_->setID(limitedRightToConfirmIDRef_);
             IDManager::instance().setID(limitedRightToConfirmIDRef_,limitedRightToConfirm_);
          }
          else if(limitedRightToConfirmNode->Attribute("href")) { limitedRightToConfirmIDRef_ = limitedRightToConfirmNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { limitedRightToConfirm_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(limitedRightToConfirmNode));}
   }

   //splitTicketNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* splitTicketNode = xmlNode->FirstChildElement("splitTicket");

   if(splitTicketNode){splitTicketIsNull_ = false;}
   else{splitTicketIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- splitTicketNode , address : " << splitTicketNode << std::endl;
   #endif
   if(splitTicketNode)
   {
      if(splitTicketNode->Attribute("href") || splitTicketNode->Attribute("id"))
      {
          if(splitTicketNode->Attribute("id"))
          {
             splitTicketIDRef_ = splitTicketNode->Attribute("id");
             splitTicket_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(splitTicketNode));
             splitTicket_->setID(splitTicketIDRef_);
             IDManager::instance().setID(splitTicketIDRef_,splitTicket_);
          }
          else if(splitTicketNode->Attribute("href")) { splitTicketIDRef_ = splitTicketNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { splitTicket_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(splitTicketNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<ManualExercise> ExerciseProcedure::getManualExercise()
{
   if(!this->manualExerciseIsNull_){
        if(manualExerciseIDRef_ != ""){
             return boost::shared_static_cast<ManualExercise>(IDManager::instance().getID(manualExerciseIDRef_));
        }else{
             return this->manualExercise_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ManualExercise>();
   }
}
boost::shared_ptr<AutomaticExercise> ExerciseProcedure::getAutomaticExercise()
{
   if(!this->automaticExerciseIsNull_){
        if(automaticExerciseIDRef_ != ""){
             return boost::shared_static_cast<AutomaticExercise>(IDManager::instance().getID(automaticExerciseIDRef_));
        }else{
             return this->automaticExercise_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<AutomaticExercise>();
   }
}
boost::shared_ptr<XsdTypeBoolean> ExerciseProcedure::getFollowUpConfirmation()
{
   if(!this->followUpConfirmationIsNull_){
        if(followUpConfirmationIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(followUpConfirmationIDRef_));
        }else{
             return this->followUpConfirmation_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<XsdTypeBoolean> ExerciseProcedure::getLimitedRightToConfirm()
{
   if(!this->limitedRightToConfirmIsNull_){
        if(limitedRightToConfirmIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(limitedRightToConfirmIDRef_));
        }else{
             return this->limitedRightToConfirm_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<XsdTypeBoolean> ExerciseProcedure::getSplitTicket()
{
   if(!this->splitTicketIsNull_){
        if(splitTicketIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(splitTicketIDRef_));
        }else{
             return this->splitTicket_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
}

